xarray/core/variable.py

```diff
@@ -307,29 +307,49 @@ def data(self, data):
     def _indexable_data(self):
         return orthogonally_indexable(self._data)
 
-    def load(self):
+    def load(self, **kwargs):
         """Manually trigger loading of this variable's data from disk or a
         remote source into memory and return this variable.
 
         Normally, it should not be necessary to call this method in user code,
         because all xarray functions should either work on deferred data or
         load data automatically.
+
+        Parameters
+        ----------
+        **kwargs : dict
+            Additional keyword arguments passed on to ``dask.array.compute``.
+
+        See Also
+        --------
+        dask.array.compute
         """
-        if not isinstance(self._data, np.ndarray):
+        if isinstance(self._data, dask_array_type):
+            self._data = np.asarray(self._data.compute(**kwargs))
```

Comment:
You don't want to invoke asarray if dask returns a scalar numpy type

Reply:
Thanks, I don't think I meant to make this change.

Reply:
To be honest, we don't define what can go in `_data` as carefully as we ought to. I guess there are two ways to define it:
- anything "array like" that defines at least `shape`, `dtype` and `__getitem__`
- what comes out of `xarray.core.variable.as_compatible_data`

Numpy scalars do actually pass through here (since they define all those attributes!)... but then would get converted into an array when calling `.values` anyways:
https://github.com/pydata/xarray/blob/78ca20a6ea1a42eb637ae2ef09189f481cfda9a2/xarray/core/variable.py#L334-L337

So I guess I agree, but on the other hand I'm also a little nervous that a dask routine might return a non-numpy scalar, which would definitely break if we don't wrap it in `asarray`. The safe thing to do is to leave this as is or call `as_compatible_data` on it.